locust/runners.py

```diff
@@ -112,6 +112,7 @@ def start_locust(_):
                     try:
                         locust().run()
                     except GreenletExit:
+                        instance.die()
```

Comment:
I suppose if we wanted the clean up code to handle regardless we could put it there. I was thinking of it more as an exception handler. I can move if you'd like.